### DISCUSSION TITLE
Wazuh Agent installation on Linux

### DISCUSSION THREAD

**Discussion by tsurara36** [wazuh/wazuh-kibana-app#5395] [Q&A] [answered]
Hi all,
I am new to Wazuh and am now trying to install an Agent on my Linux operating system. I have followed the guide to installing the agent, where i typed:
curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | gpg --no-default-keyring --keyring gnupg-ring:/usr/share/keyrings/wazuh.gpg --import && chmod 644 /usr/share/keyrings/wazuh.gpg
and it gave me
![image](https://user-images.githubusercontent.com/131188770/233104783-e8ab2319-d507-4d2f-a299-77a7205040cc.png)
After that I input
echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list
and it gave me
![image](https://user-images.githubusercontent.com/131188770/233105196-f6b7d7f5-24ef-4cc7-91b7-3e5ab653131f.png)
Finally i input
apt-get update
and it gave me
![image](https://user-images.githubusercontent.com/131188770/233105511-087dea33-07f5-4c6e-a918-3167ac8ad6eb.png)
![image](https://user-images.githubusercontent.com/131188770/233105662-416eae2b-676a-470d-9a94-3d1887785ec4.png)
What should I do to resolve this issue?

**Comment by s-ocando** [comment, marked as answer]
Hi @tsurara36, 

It seems like you have both the 3.x and the 4.x repositories on your endpoint, remove the old one to avoid conflicts.

For example, in the second command, you can remove the append ``-a`` option to overwrite the file:
```
echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main" | tee /etc/apt/sources.list.d/wazuh.list
```
Also, make sure to remove old Wazuh agents that might be present. See the [uninstalling a Wazuh agent](https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html#uninstall-a-wazuh-agent) instructions to learn more.

As an alternative to this installation method, you can follow the instructions in your Wazuh dashboard under **Agents** > **Deploy new agent**. Select your operating system and architecture, set your Wazuh manager address and use the resulting command to download, install and enroll your agent. For example:
```
curl -so wazuh-agent.deb https://packages.wazuh.com/4.x/apt/pool/main/w/wazuh-agent/wazuh-agent_4.4.1-1_amd64.deb && sudo WAZUH_MANAGER='10.0.0.2' dpkg -i ./wazuh-agent.deb
```
Let us know if this solves your question!

**Comment by tsurara36** [comment]
Hi @s-ocando,
Thanks so much for your help, I uninstalled my Wazuh Agent and reinstalled it again and it worked.
